specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/DataConnectors.json

```diff
@@ -0,0 +1,1696 @@
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "headers": {
+          "description": "The custom headers we want to add once we send request to token endpoint.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "isCredentialsInHeaders": {
+          "type": "boolean",
+          "x-nullable": true,
+          "description": "Flag indicating whether we want to send the user name and password to token endpoint in the headers."
+        },
+        "isJsonRequest": {
+          "type": "boolean",
+          "x-nullable": true,
+          "default": false,
+          "description": "Flag indicating whether the body request is JSON (header Content-Type = application/json), meaning its a Form URL encoded request (header Content-Type = application/x-www-form-urlencoded)."
+        },
+        "requestTimeoutInSeconds": {
+          "type": "integer",
+          "format": "int32",
+          "default": 100,
+          "maximum": 180,
+          "description": "Request timeout in seconds."
+        }
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "JwtToken",
+      "required": [
+        "tokenEndpoint",
+        "userName",
+        "password"
+      ]
+    },
+    "OAuthModel": {
+      "description": "Model for API authentication with OAuth2.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "properties": {
+        "authorizationCode": {
+          "type": "string",
+          "description": "The user's authorization code."
+        },
+        "clientSecret": {
+          "type": "string",
+          "description": "The Application (client) secret that the OAuth provider assigned to your app."
+        },
+        "clientId": {
+          "type": "string",
+          "description": "The Application (client) ID that the OAuth provider assigned to your app."
+        },
+        "isCredentialsInHeaders": {
+          "type": "boolean",
+          "x-nullable": true,
+          "default": false,
+          "description": "Indicating whether we want to send the clientId and clientSecret to token endpoint in the headers."
+        },
+        "scope": {
+          "type": "string",
+          "description": "The Application (client) Scope that the OAuth provider assigned to your app."
+        },
+        "redirectUri": {
+          "type": "string",
+          "description": "The Application redirect url that the user config in the OAuth provider."
+        },
+        "grantType": {
+          "type": "string",
+          "description": "The grant type, usually will be 'authorization code'."
+        },
+        "tokenEndpoint": {
+          "type": "string",
+          "description": "The token endpoint. Defines the OAuth2 refresh token."
+        },
+        "tokenEndpointHeaders": {
```

Comment:
>tokenEndpointHeaders [](http://example.com/codeflow?start=9&length=20)

ARM is trying to reduce instances of unschematized APIs to make it easier for consumers of those APIs to understand them and use them reliably, not to mention provide better support in the various SDKs.

Please make this an array or provide an explanation for why you need this.

Reply:
Same as previous comment. This discussed in Preview version and approved by ARM